test/rulesets/Base/array.jl

```diff
@@ -215,6 +221,38 @@ end
     test_rrule(fill, 3.3, (3, 3, 3))
 end
 
+@testset "filter" begin
+    @testset "Tuple" begin
+        xt10 = Tuple(vcat(0, 1, rand(8)))  # guarantee that not all > or < 0.5
+
+        # Forward
+        @test_skip test_frule(filter, >(0.5) ⊢ NoTangent(), xt10; check_inferred=false) # check_result.jl:104  Expression: ActualPrimal === ExpectedPrimal  Evaluated: NTuple{10, Float64} === NTuple{6, Float64}
```

Comment:
There are quite a few test errors, which I believe are all in CRTU not the rule. I can't mark things broken but have left them skipped. And mostly added explicit tests below.

Reply:
can we open issues on CRTU and cross link the issue number back here?

Reply:
Issues 222-233 :)